src/components/UI/Buttons/Buttons.tsx

```diff
@@ -4,16 +4,50 @@ import styles from './Buttons.module.sass';
 interface ButtonProps extends JSX.HTMLAttributes<HTMLButtonElement> {
   onClick: () => void;
   children: JSX.Element | string;
+  icon?: string; // Can't put here JSX.Element due to clash with JSX.HTMLAttributes
+  isReversed?: boolean;
+  iconAlt?: string;
 }
 
 /**
  * Button components for the UI
  */
 
 export function PrimaryButton(props: ButtonProps) {
-  return <button className={styles.primaryButton}>{props.children}</button>;
+  if (!props.icon) {
+    return <button className={styles.primaryButton}>{props.children}</button>;
+  }
+  let containerClass = styles.container;
+  if (props.isReversed) {
+    containerClass = styles.container + ' ' + styles.reversed;
+  }
+  return (
+    <button className={styles.primaryButton}>
+      <div class={containerClass}>
+        <div>{props.children}</div>
+        <img
+          class="buttonIcon"
+          src={props.icon}
+          alt={props.iconAlt ?? 'icon'}
+        />
+      </div>
+    </button>
+  );
 }
```

Comment:
Revert to the previous commit.
```suggestion
export function PrimaryButton(props: ButtonProps) {
  return <button className={styles.primaryButton}>{props.children}</button>;
}
```